[pnpm v7] vite with pnpm give " ERR_PNPM_PACKAGE_JSON_EXISTS  package.json already exists"

<!--
  If this issue affects many people in a company/big team, create a post for your company in the following discussion:
  https://github.com/pnpm/pnpm/discussions/3787
  and link the issue in your post.

  This will help us prioritize issues that affect more people.
-->

### pnpm version:
7.0.0
### Code to reproduce the issue:
`
pnpm init vite@latest
`
<!--
If there was a fatal error also include a gist of your node_modules/.pnpm-debug.log file.
-->

### Expected behavior:
project initializing steps should start
### Actual behavior:
I get this error  " ERR_PNPM_PACKAGE_JSON_EXISTS  package.json already exists" 
### Additional information:

 - `node -v` prints: v16.14.2
 - Windows, macOS, or Linux?: Linux xbuntu

--- a/.changeset/green-cows-work.md
+++ b/.changeset/green-cows-work.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/plugin-commands-init": patch
+"pnpm": patch
+---
+
+Throw an error if arguments are passed to the `pnpm init` command.

--- a/packages/plugin-commands-init/src/init.ts
+++ b/packages/plugin-commands-init/src/init.ts
@@ -25,8 +25,14 @@ export function help () {
 }
 
 export async function handler (
-  opts: Pick<UniversalOptions, 'rawConfig'>
+  opts: Pick<UniversalOptions, 'rawConfig'>,
+  params?: string[]
 ) {
+  if (params?.length) {
+    throw new PnpmError('INIT_ARG', 'init command does not accept any arguments', {
+      hint: `Maybe you wanted to run "pnpm create ${params.join(' ')}"`,
+    })
+  }
   // Using cwd instead of the dir option because the dir option
   // is set to the first parent directory that has a package.json file.
   const manifestPath = path.join(process.cwd(), 'package.json')

--- a/packages/plugin-commands-init/test/init.test.ts
+++ b/packages/plugin-commands-init/test/init.test.ts
@@ -34,3 +34,11 @@ test('init a new package.json with npmrc', async () => {
   expect(manifest.author).toBe(expectAuthor)
   expect(manifest.license).toBe(rawConfig['init-license'])
 })
+
+test('throw an error if params are passed to the init command', async () => {
+  prepare({})
+
+  await expect(
+    init.handler({ rawConfig: {} }, ['react-app'])
+  ).rejects.toThrow('init command does not accept any arguments')
+})

